arviz/data/inference_data.py

```diff
@@ -2066,21 +2066,21 @@ def concat(*args, dim=None, copy=True, inplace=False, reset_dim=True):
                         continue
 
                     # assert that variables are equal
-                    group_data = getattr(arg, group)
+                    group_data = arg[group]
                     group_vars = group_data.data_vars
 
-                    group0_data = getattr(arg0, group)
+                    group0_data = arg0[group]
                     if not inplace:
                         group0_data = deepcopy(group0_data)
                     group0_vars = group0_data.data_vars
 
                     for var in group_vars:
                         if var not in group0_vars:
-                            var_data = getattr(group_data, var)
-                            getattr(arg0, group)[var] = var_data
+                            var_data = group_data[var]
+                            arg0[group][var] = var_data
```

Comment:
I think that setitem is not implemented for Inferencedata, so this probably won't work.

Reply:
Oh, I see.

Reply:
How do I update the PR?

Reply:
this specific line should not be updated, it is already using the right `[var]` and should keep the `getattr` for the `group`. As a general rule, I think that if what is being retrieved is a group it should be getattr, [] otherwise. The use of [] and getattr is not consistent and a bit confusing, but it looks like the "index" use is consistent: `getattr(group_data, var)` is using var as "index" and should use [] instead, if using `getattr(..., group)` then getattr should be kept

Reply:
ooh, sorry. You should add a new commit to the same branch and then push. The PR will be updated with the new commit automatically